packages/client/test/end-to-end/SubscribeAll.test.ts

```diff
@@ -0,0 +1,149 @@
+        })
+        publishTestMessages = getPublishTestMessages(client, stream)
+    })
+
+    afterEach(async () => {
+        // @ts-expect-error private
+        expect(await client.subscriber.count()).toBe(0)
+        // @ts-expect-error private
+        expect(await client.subscriber.count(stream.id)).toBe(0)
+        // @ts-expect-error private
+        expect(client.subscriber.countSubscriptionSessions()).toBe(0)
+    })
+
+    afterEach(async () => {
+        await wait(0)
+        await client?.destroy()
+        // ensure no unexpected errors
+        expect(onError).toHaveBeenCalledTimes(expectErrors)
+    })
+
+    it('subscribes to all partitions', async () => {
+        const subMsgs: any[] = []
+        await client.subscribeAll(stream.id, (msg) => {
+            subMsgs.push(msg)
+        })
+        const eachPartition = Array(PARTITIONS).fill(0).map((_v, streamPartition) => streamPartition)
+        const pubs = await Promise.all(eachPartition.map((streamPartition) => {
+            return publishTestMessages(NUM_MESSAGES, { partitionKey: streamPartition })
+        }))
+        const publishedMsgs = pubs.flat()
+        expect(publishedMsgs.length).toBe(PARTITIONS * NUM_MESSAGES)
+        await waitForCondition(() => subMsgs.length >= (PARTITIONS * NUM_MESSAGES), 25000)
+        for (const msg of publishedMsgs) {
+            expect(subMsgs).toContainEqual(msg)
+        }
+        await client.unsubscribe()
+        expect(client.countAll()).toBe(0)
+    })
+
+    it('can stop prematurely', async () => {
+        const subMsgs: any[] = []
+        const sub = await client.subscribeAll(stream.id, (msg) => {
+            subMsgs.push(msg)
+            if (subMsgs.length === MAX_MESSAGES) {
+                sub.return()
+            }
+        })
+        const eachPartition = Array(PARTITIONS).fill(0).map((_v, streamPartition) => streamPartition)
+        const pubs = await Promise.all(eachPartition.map((streamPartition) => {
+            return publishTestMessages(NUM_MESSAGES, { partitionKey: streamPartition })
+        }))
+        const publishedMsgs = pubs.flat()
+        expect(publishedMsgs.length).toBe(PARTITIONS * NUM_MESSAGES)
+        await sub.onFinally()
+        // got the messages
+        expect(subMsgs).toHaveLength(MAX_MESSAGES)
+        // unsubscribed from everything
+        expect(client.countAll()).toBe(0)
+    })
+
+    it('stops with unsubscribeAll', async () => {
+        const subMsgs: any[] = []
+        const sub = await client.subscribeAll(stream.id, (msg) => {
+            subMsgs.push(msg)
+            if (subMsgs.length === MAX_MESSAGES) {
+                client.unsubscribe()
+            }
+        })
+        const eachPartition = Array(PARTITIONS).fill(0).map((_v, streamPartition) => streamPartition)
+        const pubs = await Promise.all(eachPartition.map((streamPartition) => {
+            return publishTestMessages(NUM_MESSAGES, { partitionKey: streamPartition })
+        }))
```

Comment:
I noticed this same (similar?) segment is repeated (also present in ResendAll test?), extract into helper function?

Reply:
Switched to https://lodash.com/docs/4.17.15#range